core/mastodon/src/test/java/br/com/orcinus/orca/core/mastodon/notification/NotificationServiceTests.kt

```diff
@@ -0,0 +1,199 @@
+    ) {
+      val service = NotificationService(requester, authenticationLock, coroutineContext)
+      val intent = Intent(context, NotificationService::class.java)
+      val controller = ServiceController.of(service, intent)
+      val messageBuilder = RemoteMessage.Builder(MESSAGE_RECIPIENT)
+      assertThat(MastodonNotification.Type.entries).each { typeAssert ->
+        controller.bind()
+        typeAssert.given { type ->
+          val dto =
+            MastodonNotification(
+              /* id = */ "${type.ordinal}",
+              type,
+              dtoCreatedAt,
+              MastodonAccount.default,
+              MastodonStatus.default
+            )
+          val message = messageBuilder.setData(dto.toMap()).build()
+
+          fun hasNotificationNotBeenSent(): Boolean {
+            return service.notificationManager.activeNotifications
+              .map(StatusBarNotification::getId)
+              .contains(dto.normalizedID)
+              .not()
+          }
+
+          service.onMessageReceived(message)
+          @Suppress("ControlFlowWithEmptyBody") while (hasNotificationNotBeenSent()) {}
+          assertThat(service)
+            .prop(NotificationService::notificationManager)
+            .prop(NotificationManager::getActiveNotifications)
+            .transform("of $type") { statusBarNotifications ->
+              statusBarNotifications.find { statusBarNotification ->
+                statusBarNotification.id == dto.normalizedID
+              }
+            }
+            .isNotNull()
+            .prop(StatusBarNotification::getNotification)
+            .all {
+              prop(Notification::getChannelId).isEqualTo(type.channelID)
+              transform("title") { it.extras.getString(Notification.EXTRA_TITLE) }
+                .isEqualTo(type.getContentTitleAsync(context, authenticationLock, dto).get())
+            }
+        }
+        controller.unbind()
+      }
+    }
+  }
+
+  @Test
+  fun cancelsSentNotificationsWhenUnbound() {
+    runAuthenticatedRequesterTest {
+      val service = NotificationService(requester, authenticationLock, coroutineContext)
+      val intent = Intent(context, NotificationService::class.java)
+      val controller =
+        ServiceController.of(service, intent).apply(ServiceController<NotificationService>::bind)
+      MastodonNotification.Type.entries
+        .map {
+          MastodonNotification(
+            /* id = */ "0",
+            /* type = */ it,
+            dtoCreatedAt,
+            MastodonAccount.default,
+            MastodonStatus.default
+          )
+        }
+        .map { it.normalizedID to it.toNotification(context, authenticationLock) }
+        .forEach { (id, notification) -> service.notificationManager.notify(id, notification) }
+      controller.unbind()
+      assertThat(service)
+        .prop(NotificationService::notificationManager)
+        .prop(NotificationManager::getActiveNotifications)
+        .isNotEmpty()
```

Comment:
`cancelsSentNotificationsWhenUnbound`? `assertThat…isNotEmpty()`? 🙃

If they have been cancelled, shouldn't there be no active notifications at all (in which case they'd be empty)?

Reply:
Even sadder is the fact that the test is passing just fine 🙂👊🏽